pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractJavaRule.java

```diff
@@ -13,9 +13,16 @@
 import net.sourceforge.pmd.lang.java.ast.*;
 import net.sourceforge.pmd.lang.rule.AbstractRule;
 import net.sourceforge.pmd.lang.rule.ImmutableLanguage;
+import net.sourceforge.pmd.properties.StringMultiProperty;
 
 public abstract class AbstractJavaRule extends AbstractRule implements JavaParserVisitor, ImmutableLanguage {
 
+    protected StringMultiProperty.StringMultiPBuilder ignoredAnnotBuilder
+        = StringMultiProperty.named("ignoredAnnotations")
+        .desc("Fully qualified names of the annotation types that should be ignored by this rule");
+
+    protected StringMultiProperty ignoredAnnotDescriptor;
```

Comment:
I'd have the property be created here directly. Just add a method that returns the default values, have a default implementation here (returning nothing at all), and let subclasses override it.

We should register the property in this class' constructor too

Reply:
> I'd have the property be created here directly.

Yes, I tried that. But `defaultValue` field in property is final. If the property was created here directly, subclass can't modify the default values.

> Just add a method that returns the default values, have a default implementation here (returning nothing at all), and let subclasses override it.

IMO, processing default values is task of `Property` class. So we shouldn't do it in `Rule` class.

Reply:
I think I'm not making myself clear... what's wrong with doing this?

```java
protected final StringMultiProperty ignoredAnnotDescriptor = StringMultiProperty.named("...").defaultValue(defaultSuppressionAnnotations())...; 

protected Collection<String> defaultSuppressionAnnotations() {
  return Collections.emptyList();
}
```

Subclasses can simply override `defaultSuppressionAnnotations` and return their own defaults

Reply:
I get it now:)